shieldapi/__init__.py

```diff
@@ -0,0 +1,37 @@
+import logging
+import os
+
+# Configure logging
+logging.basicConfig(level=logging.INFO)
+logger = logging.getLogger("shieldapi")
+
+
+def check_mandatory_env_vars():
+    """
+    Checks if all mandatory environment variables are set.
+    Skips the check if SKIP_ENV_CHECK is set.
+    Raises an EnvironmentError if any are missing.
+    """
+    # Allow skipping the check if SKIP_ENV_CHECK is set
+    if os.environ.get("SKIP_ENV_CHECK"):
```

Comment:
Did you test this again True and False values? os.environ.get always returns a `str` and does not apply any type casting. The truth of a `str` always returns `True` which leads to the behaviour that this env does not deliver the required bypass, no matter what the value of `SKIP_ENV_CHECK` is.

Reply:
Making a string matching with `["1", "true", "yes"]` might lead to unexpected behaviour especially when the available options are not documented or trailing whitespaces are accidentally read from env.

`literal_eval` is a safer way:
```{python}
from ast import literal_eval

skip_check = os.environ.get("SKIP_ENV_CHECK", "True")
try:
    skip_check = literal_eval(skip_check)
except ValueError:
    raise ValueError(f"Not a boolean value for `SKIP_ENV_CHECK`. {skip_check)") 
```

Reply:
Adapted